### DISCUSSION TITLE
Update data of single Item from BagPart

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#10780] [Q&A] [answered]
How can i update single item from BagPart?

**Comment by Skrypt** [comment, marked as answer]
Good Question. Each Named Part or BagPart has an `IEnumerable<ContentItem>`. So for example a Named Part named "Team" that contains content items of a Content Type named "TeamMember" would contain an IEnumerable<ContentItem> of "TeamMember".

https://github.com/OrchardCMS/OrchardCore/blob/72fab98a63f2f98b99a3fd2981e1a46df6ba78bf/src/OrchardCore.Modules/OrchardCore.Flows/Models/BagPart.cs#L10

Then, if you want to update the BagPart you can look at the BagPartDisplayDriver to see how we update it.

https://github.com/OrchardCMS/OrchardCore/blob/72fab98a63f2f98b99a3fd2981e1a46df6ba78bf/src/OrchardCore.Modules/OrchardCore.Flows/Drivers/BagPartDisplayDriver.cs#L68-L84

One thing to remember is that we can't actually find a BagPart ContentItem by doing a GetContentItemById because this BagPart JSON data is contained inside its "Parent" Content Item. So, you can find the parent content item by ContentItemId and then you find the BagPart ContentItems by listing them. Then, you can do a LINQ query to find a specific ContentItemId that you would want to update in that List. 

`var existingContentItem = part.ContentItems.FirstOrDefault(x => String.Equals(x.ContentItemId, model.Prefixes[i], StringComparison.OrdinalIgnoreCase));`

So as you can see above when we update an existing ContentItem we do simply : `contentItem.Merge(existingContentItem); `

Then of course last step would be to `_session.save(parentContentItem)` the parent content Item.

**Reply by vitalybrandes** [reply]
@Skrypt  Did i get it right? We cannot update single item in the BagPart, we need to rewrite all items?
